pkg/bucket/bucket_controller.go

```diff
@@ -100,15 +106,18 @@ func (b *BucketListener) Add(ctx context.Context, inputBucket *v1alpha1.Bucket)
 		bucketID = bucket.Spec.ExistingBucketID
 		if bucket.Spec.Parameters == nil {
 			bucketClass, err := b.bucketClasses().Get(ctx, bucket.Spec.BucketClassName, metav1.GetOptions{})
-			if err != nil {
+			if kubeerrors.IsNotFound(err) {
+				b.recordEvent(inputBucket, v1.EventTypeWarning, events.ProvisioningFailed, "BucketClass provided in the BucketClaim does not exist")
```

Comment:
Since the controller knows the bucketClass name, it seems best to send that info to users so they can fix the issue more quickly and not have to cross reference as many details.

```suggestion
				b.recordEvent(inputBucket, v1.EventTypeWarning, events.ProvisioningFailed, "BucketClass %q provided in the BucketClaim does not exist", bucket.Spec.BucketClassName)
```